app/packages/looker/src/worker/canvas-decoder.ts

```diff
@@ -67,7 +69,12 @@ export const recastBufferToMonoChannel = (
   return uint8Array.slice(0, totalPixels).buffer;
 };
 
-export const decodeWithCanvas = async (blob: Blob, cls: string) => {
+export const decodeWithCanvas = async (
+  blob: Blob,
+  cls: string,
+  field: string,
+  coloring: Coloring
+) => {
```

Comment:
This could help `decodeWithCanvas` responsibilities to a minimum
```suggestion
export const decodeWithCanvas = async (
  blob: Blob,
  cls: string,
  isRgb: boolean
) => {
```

Reply:
I update some of this in https://github.com/voxel51/fiftyone/pull/5413